lib/org/chromium/webidl/LocalGitRunner.js

```diff
@@ -115,38 +105,30 @@ foam.CLASS({
   ],
 
   methods: [
-    function init() {
-      this.validate();
-      this.ensureFetched_().then(this.onFetched);
-    },
-    function put(obj) {
-      return Promise.reject('LocalGitIDLFileDAO: DAO is read-only.');
-    },
-    function remove(obj) {
-      return Promise.reject('LocalGitIDLFileDAO: DAO is read-only.');
-    },
-    function find(objOrId) {
-      return this.checkNotReady_() || this.delegate.find(objOrId);
-    },
-    function select(sink, skip, limit, order, predicate) {
-      return this.checkNotReady_() ||
-          this.delegate.select(sink, skip, limit, order, predicate);
-    },
-    function removeAll(skip, limit, order, predicate) {
-      return Promise.reject('LocalGitIDLFileDAO: DAO is read-only.');
-    },
+    function run(args) {
+      if (!args || !args.config) throw new Error("LocalGitRunner: Missing configuration file!");
+      var config = args.config;
 
-    {
-      name: 'checkNotReady_',
-      documentation: `Return an InternalException promise rejection iff DAO is
-        not ready to accept operations yet. Otherwise, return false.`,
-      code: function() {
-        if (this.isReady_) return false;
+      // Verify that required properties are present
+      if (!config.repositoryURL || !config.localRepositoryPath || !config.idlFileContentsFactory)
+        throw new Error("LocalGitRunner: Invalid configuration file");
 
-        // Reject on internal exception when not ready. By default, this will be
-        // retried by a StoreAndForwardDAO.
-        return Promise.reject(this.InternalException.create());
-      },
+      for (var prop in config) {
+        this[prop] = config[prop];
```

Comment:
This feels unsafe. It could override things like this.childProcess_ to inject a new execSync implementation that does something different. The code should probably either copy the properties that matter, or store to config object and access properties on it.